src/DynamoCoreWpf/ViewModels/PackageManager/InstalledPackagesViewModel.cs

```diff
@@ -1,51 +1,257 @@
-        public ObservableCollection<PackageViewModel> LocalPackages { get { return _localPackages; } }
+        /// <summary>
+        /// The current filtered list of local packages
+        /// </summary>
+        public ObservableCollection<PackageViewModel> LocalPackages { get; } = new ObservableCollection<PackageViewModel>();
+
+        /// <summary>
+        /// Possible filters to use for the filtered list of packages
+        /// </summary>
+        public ObservableCollection<PackageFilter> Filters { get; } = new ObservableCollection<PackageFilter>();
 
         private readonly DynamoViewModel dynamoViewModel;
+
+        /// <summary>
+        /// Back pointer to the Package Loader
+        /// </summary>
         public PackageLoader Model { get; private set; }
 
+        /// <summary>
+        /// Create a new view model for installed packages
+        /// </summary>
+        /// <param name="dynamoViewModel">Back pointer to the dynamo view model</param>
+        /// <param name="model">Back pointer to the package loader</param>
         public InstalledPackagesViewModel(DynamoViewModel dynamoViewModel, PackageLoader model)
         {
             this.Model = model;
             this.dynamoViewModel = dynamoViewModel;
-            
+
             InitializeLocalPackages();
+            PopulateFilters();
+        }
+
+        /// <summary>
+        /// Filter the list of local packages based on the current filter selection
+        /// </summary>
+        internal void ApplyPackageFilter()
+        {
+            ClearPackagesViewModel();
+
+
+            if (NoActiveFilterSelected)
+            {
+                LocalPackages.AddRange(Model.LocalPackages.Select(NewPackageViewModel));
+            }
+            else
+            {
+                var currentFilter = CurrentFilterSelection;
+                foreach (var pkg in Model.LocalPackages)
+                {
+                    var viewModel = NewPackageViewModel(pkg);
+                    if (viewModel.PackageLoadStateText == currentFilter)
+                    {
+                        LocalPackages.Add(viewModel);
+                    }
+                    else
+                    {
+                        ClearPackageViewModel(viewModel);
+                    }
+                }
+            }
+            RaisePropertyChanged(nameof(LocalPackages));
         }
 
-        internal void GoToWebsite()
+        /// <summary>
+        /// Populate the list of available filters based on the packages currently known by the dynamo view model
+        /// </summary>
+        internal void PopulateFilters()
         {
-            dynamoViewModel.PackageManagerClientViewModel.GoToWebsite();
+            var currentSelection = CurrentFilterSelection;
+            Filters.Clear();
+            var filterNames = Model.LocalPackages.Select(pkg => new PackageViewModel(dynamoViewModel, pkg))
```

Comment:
Why don;t you use the `public ObservableCollection<PackageViewModel> LocalPackages` here, instead of calling `Model.LocalPackages.Select(pkg => new PackageViewModel(dynamoViewModel, pkg))` ?

Reply:
LocalPackages may be filtered when we get here and only contain a subset of all packages.